include/graaflib/algorithm/utils.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <graaflib/graph.h>
+
+namespace graaf {
+
+template <typename VERTEX_T, typename EDGE_T>
+directed_graph<VERTEX_T, EDGE_T> get_transposed_graph(
```

Comment:
[very minor]: In the codebase we have the habit of splitting definition and implementation. Maybe we can add a `utils.tpp` file for consistency. I would however also be fine to do this in a separate PR as the current one unblocks another PR.